src/algebra/algebra/basic.lean

```diff
@@ -1195,6 +1199,51 @@ by { ext, simp only [linear_map.id_coe, mul_one, id.def, lmul_right_apply] }
   lmul_right R (a * b) = (lmul_right R b).comp (lmul_right R a) :=
 by { ext, simp only [lmul_right_apply, linear_map.comp_apply, mul_assoc] }
 
+@[simp] lemma lmul_left_zero_eq_zero :
+  lmul_left R (0 : A) = 0 :=
+by { ext, simp, }
+
+@[simp] lemma lmul_right_zero_eq_zero :
+  lmul_right R (0 : A) = 0 :=
+by { ext, simp, }
+
+@[simp] lemma lmul_left_eq_zero_iff (a : A) :
+  lmul_left R a = 0 ↔ a = 0 :=
+begin
+  split; intros h,
+  { rw [← mul_one a, ← lmul_left_apply a 1, h, linear_map.zero_apply], },
+  { rw h, exact lmul_left_zero_eq_zero, },
+end
+
+@[simp] lemma lmul_right_eq_zero_iff (a : A) :
+  lmul_right R a = 0 ↔ a = 0 :=
+begin
+  split; intros h,
+  { rw [← one_mul a, ← lmul_right_apply a 1, h, linear_map.zero_apply], },
+  { rw h, exact lmul_right_zero_eq_zero, },
+end
+
+@[simp] lemma pow_lmul_left (a : A) (n : ℕ) :
+  (lmul_left R a)^n = lmul_left R (a ^n) :=
+begin
+  ext b,
+  induction n with n ih,
+  { simp only [linear_map.id_coe, linear_map.one_apply, lmul_left_one, id.def, pow_zero], },
+  { simp only [lmul_left_apply] at ih,
+    simp only [lmul_left_apply, pow_succ, linear_map.mul_apply, ih, mul_assoc], },
+end
+
+@[simp] lemma pow_lmul_right (a : A) (n : ℕ) :
+  (lmul_right R a)^n = lmul_right R (a ^n) :=
+begin
+  ext b,
+  induction n with n ih,
+  { simp only [linear_map.id_coe, linear_map.one_apply, lmul_right_one, id.def, pow_zero], },
+  { rw lmul_right_apply at ih,
+    rw [pow_succ, pow_succ, linear_map.mul_apply, ih, lmul_right_apply, lmul_right_apply,
+      mul_assoc, ← pow_succ, ← pow_succ'], },
+end
```

Comment:
Slight shorter with an import of `algebra.iterate_hom`:
```
begin
  apply linear_map.coe_injective,
  dsimp only, -- TODO: remove this by fixing `coe_injective` to not introduce junk
  rw [linear_map.coe_pow],
  exact mul_right_iterate a n,
end
```

Reply:
Nice!